src/esp/physics/PhysicsManager.cpp

```diff
@@ -103,9 +103,17 @@ int PhysicsManager::addObject(const std::string& configFileHandle,
 
   //! Draw object via resource manager
   //! Render node as child of physics node
-  resourceManager_.addObjectToDrawables(
-      configFileHandle, existingObjects_.at(nextObjectID_)->visualNode_,
-      drawables, lightSetup);
+  std::vector<scene::SceneNode*> newNodes =
+      resourceManager_.addObjectToDrawables(
+          configFileHandle, existingObjects_.at(nextObjectID_)->visualNode_,
+          drawables, lightSetup);
+
+  existingObjects_.at(nextObjectID_)
+      ->visualNodes_.push_back(existingObjects_.at(nextObjectID_)->visualNode_);
+  existingObjects_.at(nextObjectID_)
+      ->visualNodes_.insert(
+          existingObjects_.at(nextObjectID_)->visualNodes_.end(),
+          newNodes.begin(), newNodes.end());
```

Comment:
Hmm, actually here the `*Internal` could be used too:

```suggestion
  // done first to have the order the same as before
  existingObjects_.at(nextObjectID_)
      ->visualNodes_.push_back(existingObjects_.at(nextObjectID_)->visualNode_);

  resourceManager_.addObjectToDrawablesInternal(
      configFileHandle, existingObjects_.at(nextObjectID_)->visualNode_,
      drawables, lightSetup, existingObjects_.at(nextObjectID_)->visualNodes_);
```

Of course now it needs a better name than `*Internal` ... `*Into`? `*Out`?